Applications/Spire/Source/OrderImbalanceIndicator/CachedOrderImbalanceIndicatorModel.cpp

```diff
@@ -0,0 +1,106 @@
+#include "Spire/OrderImbalanceIndicator/CachedOrderImbalanceIndicatorModel.hpp"
+
+using namespace boost;
+using namespace boost::icl;
+using namespace boost::posix_time;
+using namespace Nexus;
+using namespace Spire;
+
+CachedOrderImbalanceIndicatorModel::CachedOrderImbalanceIndicatorModel(
+    std::shared_ptr<OrderImbalanceIndicatorModel> source)
+    : m_source_model(std::move(source)) {
+  auto [connection, promise] = m_source_model->subscribe(
```

Comment:
Should be able to use something like:

    std::tie(m_subscription_connection, m_subscription_promise) = m_source_model->...

Reply:
It looks like this would required adding a std::tuple<...> operator:

https://stackoverflow.com/a/26278338/11234391

Reply:
With the example you showed me above, the compiler complains that std::tuple<connection, QtPromise<...>> is incompatible with SubscriptionResult<...>.

Reply:
Yes that's correct, ignore this.

Reply:
Do you want me to leave this or add the operator?

Reply:
Sorry, didn't get the update.